unity/Assets/UnitTests/TestThirdPartyCameraAndMainCamera.cs

```diff
@@ -312,5 +312,329 @@ public IEnumerator TestUpdateMainCamera()
+            // Debug.Log($"parent relative camera rot: {metadata.thirdPartyCameras[0].parentPositionRelativeThirdPartyCameraRotation:F10}");
+            // Debug.Log($"parent object name: {metadata.thirdPartyCameras[0].parentObjectName}");
+
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraPosition.x,
+                -4.0000010000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraPosition.y,
+                2.9009990000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraPosition.z,
+                2.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraRotation.x,
+                20.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraRotation.y,
+                290.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].worldRelativeThirdPartyCameraRotation.z,
+                20.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].agentPositionRelativeThirdPartyCameraPosition.Value.x,
+                1.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].agentPositionRelativeThirdPartyCameraPosition.Value.y,
+                2.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].agentPositionRelativeThirdPartyCameraPosition.Value.z,
+                3.0000020000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].agentPositionRelativeThirdPartyCameraRotation.Value.x,
+                20.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].agentPositionRelativeThirdPartyCameraRotation.Value.y,
+                20.0000000000f
+            );
+            Assert.AreEqual(result, true);
+            result = Mathf.Approximately(
+                metadata.thirdPartyCameras[0].agentPositionRelativeThirdPartyCameraRotation.Value.z,
```

Comment:
Change variable name to `agentRotationRelativeThirdPartyCameraRotation`

Reply:
updated in ef3bfb20c08ec9a7a991b23d7e9008af3c72d132